class/defaults.yml

```diff
@@ -1,5 +1,11 @@
 parameters:
   crossplane:
+    distribution: default
```

Comment:
Not having `distribution` set to the cluster's distribution fact (`${facts:distribution}`) by default feels counter-intuitive. 

Additionally, there's a high chance that someone will set this parameter to `${facts:distribution}` somewhere causing the component to break in unexpected ways for clusters which have a distribution which isn't listed in the lookup table.

For those reasons, I don't think using lookup tables is the right choice here. Instead, if you want to handle the changes required for OpenShift in the component, I would instead suggest writing a postprocessing filter which removes the `runAsUser` field from the manifests rendered from the Helm chart.

Alternatively, you could add logic to the component to deploy an SCC which allows using UID 66532 and RBAC rules which grant the Crossplane components the right to use the SCC. Of course you'd have to do this only when `${facts:distribution}` is `openshift4`.

Reply:
After careful consideration the post processing approach seemed the best, that's why I move into that direction.